vm/src/pyobject.rs

```diff
@@ -788,9 +791,30 @@ pub trait IdProtocol {
     }
 }
 
-impl IdProtocol for PyObjectRef {
+#[derive(Debug)]
+enum Never {}
```

Comment:
What's the use of this?

Reply:
We need a concrete payload type to cast a fat pointer (`*const PyObject<dyn PyObjectPayload>`) to a thin pointer (`*const PyObject<SomeConcreteType>`) so we can use its address for the object id. The ideal type would be the never type `!`, but that's not stable, so creating an enum with no variants is the standard workaround.